src/m365/spo/commands/file/file-retentionlabel-remove.spec.ts

```diff
@@ -0,0 +1,263 @@
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    sinon.stub(Cli, 'prompt').callsFake(async (options: any) => {
+      promptOptions = options;
+      return { continue: false };
+    });
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get,
+      Cli.prompt,
+      Cli.executeCommandWithOutput
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.FILE_RETENTIONLABEL_REMOVE), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('prompts before removing retentionlabel from a file when confirmation argument not passed', async () => {
+    await command.action(logger, { options: { debug: false, webUrl: webUrl, fileUrl: fileUrl } });
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing file retention label when prompt not confirmed', async () => {
+    const postSpy = sinon.spy(request, 'delete');
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: false }
+    ));
+    await command.action(logger, {
+      options: {
+        debug: false,
+        fileUrl: fileUrl,
+        webUrl: webUrl
+      }
+    });
+    assert(postSpy.notCalled);
+  });
+
+  it('removes the retentionlabel from a file based on fileUrl when prompt confirmed', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/_api/web/GetFileByServerRelativeUrl('${formatting.encodeQueryParameter(fileUrl)}')?$expand=ListItemAllFields`) {
+        return { ListItemAllFields: { Id: listId }, ServerRelativeUrl: fileUrl };
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: true }
+    ));
+
+    const postSpy = sinon.stub(Cli, 'executeCommandWithOutput').callsFake(async (command): Promise<any> => {
+      if (command === SpoListItemRetentionLabelRemoveCommand) {
+        return ({
+          stdout: SpoListItemRetentionLabelRemoveCommandOutput
+        });
+      }
+
+      throw new CommandError('Unknown case');
+    });
+
+    await command.action(logger, {
+      options: {
+        debug: false,
+        fileUrl: fileUrl,
+        webUrl: webUrl
+      }
+    });
+    assert(postSpy.called);
```

Comment:
I don't think asserting that the `postSpy` is called is the corret way to validate that the command ran fine. I believe yo can wrap the command.action execution with await `assert.doesNotReject` or something like that. You need to validate that the command does not throw an error and executes fine.